File: module-core/src/main/java/maple/expectation/core/calculator/v4/impl/BaseEquipmentItem.java
package maple.expectation.core.calculator.v4.impl;

import java.math.BigDecimal;
import java.util.Optional;
import lombok.RequiredArgsConstructor;
import maple.expectation.core.calculator.v4.EquipmentExpectationCalculator;

/**
 * V4 기본 장비 아이템 (Decorator Pattern - Concrete Component) (#240)
 *
 * <h3>역할</h3>
 *
 * <p>Decorator 체인의 시작점. 기본 아이템 자체의 비용은 0입니다.
 *
 * @see EquipmentExpectationCalculator 대상 인터페이스
 */
@RequiredArgsConstructor
public class BaseEquipmentItem implements EquipmentExpectationCalculator {

  private final String itemName;
  private final int itemLevel;
  private final int currentStar;

  @Override
  public BigDecimal calculateCost() {
    return BigDecimal.ZERO; // 기본 아이템 자체의 비용은 0
  }

  @Override
  public String getEnhancePath() {
    return itemName;
  }

  @Override
  public Optional<BigDecimal> getTrials() {
    return Optional.of(BigDecimal.ZERO);
  }

  @Override
  public CostBreakdown getDetailedCosts() {
    return CostBreakdown.empty();
  }

  public int getItemLevel() {
    return itemLevel;
  }

  public int getCurrentStar() {
    return currentStar;
  }
}

File: module-core/src/main/kotlin/maple/expectation/core/calculator/domain/BaseEquipmentItem.kt
package maple.expectation.core.calculator.domain

import java.math.BigDecimal
import java.util.Optional

/**
 * Core Domain: V4 기본 장비 아이템
 *
 * Decorator Pattern (GoF) - Concrete Component
 * Decorator 체인의 시작점. 기본 아이템 자체의 비용은 0입니다.
 *
 * Port-Based Architecture (ADR-004):
 * - module-core의 순수 도메인 모델
 * - Spring @Component 제거
 * - 불변 데이터 클래스 (data class)
 *
 * @see EquipmentExpectationCalculatorPort 대상 인터페이스
 * @see EquipmentEnhanceDecorator 추상 장식자
 */
data class BaseEquipmentItem(
    private val itemName: String,
    private val itemLevel: Int,
    private val currentStar: Int
) : EquipmentExpectationCalculatorPort {

    override fun calculateCost(): BigDecimal = BigDecimal.ZERO

    override fun getEnhancePath(): String = itemName

    override fun getTrials(): Optional<BigDecimal> = Optional.of(BigDecimal.ZERO)

    override fun getDetailedCosts(): EquipmentExpectationCalculatorPort.CostBreakdown =
        EquipmentExpectationCalculatorPort.CostBreakdown.empty()

    fun getItemLevel(): Int = itemLevel

    fun getCurrentStar(): Int = currentStar
}
